ww3tools/ProcSat_interpolation.py

```diff
@@ -0,0 +1,515 @@
+    })
+
+    # Create DataArrays for all data with consistent time representation and units
+    hs_dataarray = xr.DataArray(hs.values, coords={'time': satellite_time_unix}, dims=['time'], name='hs_time_averaged_satellite')
+    hs_dataarray.attrs['units'] = 'm'
+    wsp_dataarray = xr.DataArray(wsp.values, coords={'time': satellite_time_unix}, dims=['time'], name='wsp_time_averaged_satellite')
+    wsp_dataarray.attrs['units'] = 'm/s'
+    interpolated_htsgw_surface = xr.DataArray(interpolated_htsgw_surface_values, coords={'time': satellite_time_unix}, dims=['time'], name='HTSGW_surface_interpolated')
+    interpolated_htsgw_surface.attrs['units'] = 'm'
+    interpolated_wind_surface = xr.DataArray(interpolated_wind_surface_values, coords={'time': satellite_time_unix}, dims=['time'], name='WIND_surface_interpolated')
+    interpolated_wind_surface.attrs['units'] = 'm/s'
+    longitude_dataarray = xr.DataArray(filtered_satellite_longitudes, coords={'time': satellite_time_unix}, dims=['time'], name='longitude')
+    longitude_dataarray.attrs['units'] = 'degrees_east'
+    latitude_dataarray = xr.DataArray(filtered_satellite_latitudes, coords={'time': satellite_time_unix}, dims=['time'], name='latitude')
+    latitude_dataarray.attrs['units'] = 'degrees_north'
+    hs_cal_dataarray = xr.DataArray(hs_cal.values, coords={'time': satellite_time_unix}, dims=['time'], name='hs_cal_time_averaged_satellite')
+    hs_cal_dataarray.attrs['units'] = 'm'
+    wsp_cal_dataarray = xr.DataArray(wsp_cal.values, coords={'time': satellite_time_unix}, dims=['time'], name='wsp_cal_time_averaged_satellite')
+    wsp_dataarray.attrs['units'] = 'm/s'
+    fcst_hr_dataarray = xr.DataArray(fcst_hr2_ws, coords={'time': satellite_time_unix}, dims=['time'], name='fcst_time').assign_attrs(units='hours') #fcst_hr
+
+
+    # Combine into a single dataset / 'time': satellite_time_unix
+    interpolated_dataset = xr.Dataset({
+        'time': time_dataarray,
+        'HTSGW_surface_interpolated': interpolated_htsgw_surface,
+        'WIND_surface_interpolated': interpolated_wind_surface,
+        'hs_time_averaged_satellite': hs_dataarray,
+        'wsp_time_averaged_satellite': wsp_dataarray,
+        'longitude': longitude_dataarray,
+        'latitude': latitude_dataarray,
+        'hs_cal_time_averaged_satellite':hs_cal_dataarray,
+        'wsp_cal_time_averaged_satellite':wsp_dataarray,
+        'fcst_hr': fcst_hr_dataarray
+    })
+
+    # Before saving your dataset, add the reference times as attributes
+    interpolated_dataset.attrs['satellite_name'] = satellite_name
+    interpolated_dataset.attrs['reference_time'] = reference_time
+#    interpolated_dataset.attrs['reference_time'] = f"{reference_time} seconds since 1970-01-01 00:00:00"
+    interpolated_dataset.attrs['reference_time'] = reference_time_unix  # Store as numeric value
+    interpolated_dataset.attrs['reference_time_units'] = "seconds since 1970-01-01 00:00:00"  # Specify units separately
```

Comment:
This is missing from the grib2 output, can it be added?  

Also, the output names between grib2 and netcdf are inconsistent.  While it would be ideal to have only one place where we are writing out output, can we at least have consistency in names of variables between the nc and grib2 in terms of output variable names so that we do not have to account for this in scripts that read in output from this code?

Reply:
The names are consistent now, and the unit for the initial_condition_time is added.

Reply:
Is reference_time_units clear to everyone or would initial_condition_time_units be more obvious?